consul/query_endpoint.go

```diff
@@ -0,0 +1,454 @@
+package consul
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/hashicorp/consul/consul/structs"
+)
+
+var (
+	// ErrQueryNotFound is returned if the query lookup failed.
+	ErrQueryNotFound = errors.New("Query not found")
+)
+
+// Query manages the prepared query endpoint.
+type Query struct {
```

Comment:
Similarly, I think "PreparedQuery" or just "Prepared" would be less confusing given that "Query" is so overloaded